### ISSUE TITLE
Can't parse GNU tar file with long paths

### ISSUE THREAD

**Issue by lnicola** [Keruspe/tar.rs#5]
See the attached file. It was created using GNU `tar`:

```
$ touch 01234567801234567801234567801234567801234567801234567801234567801234567801234567899999999901234567890123456780123456780123456780123456780123456780123456780123456780123456780123456789999999990123456789
$ tar cf test.tar 01234567801234567801234567801234567801234567801234567801234567801234567801234567899999999901234567890123456780123456780123456780123456780123456780123456780123456780123456780123456789999999990123456789
```

[test.zip](https://github.com/Keruspe/tar-parser.rs/files/373089/test.zip)

**Comment by Keruspe** [comment]
Hi,

Thanks for the report, will look into it

**Comment by Keruspe** [comment]
Ok, so ftr, this tar archive has an entry of type 'L' (which is in the "vendor-specific" reserved range which are not implemented yet), followed by an entry of type '2' which is supposed to be a symlink ubut I guess that's just some metadata for the 'L' one, and then an entry of type '0' which is the actual file.

I'm gonna look at gnu tar sources to get how 'L' is working, implement that and probably error-out when we encounter an entry with a non-implemented type.

**Comment by lnicola** [comment]
Thanks for looking into it. It's a GNU extension used to implement long file names. The extra entry contains the real name (followed by a `NULL`, I think).

**Comment by Keruspe** [comment]
Sorry for the delay, this should be fixed in 0.3.0
